docs_src/design/adr/core/0021-Device-Profile-Changes.md

```diff
@@ -0,0 +1,66 @@
+# Changes to Device Profiles
+
+## Status
+** Proposed **
+(as of 1/20/22)
+
+Please see a [prior PR](https://github.com/edgexfoundry/edgex-docs/pull/605) on this topic that detailed much of the debate and context on this issue.  For clarity and simplicity, that PR was closed in favor of this simpler ADR.
+
+## Context
+While the device profile has always been the way to describe a device/sensor and template its communications to the rest of the EdgeX platform, over the course of EdgeX evolution there have been changes in what could change in a profile (often based on its associations to other EdgeX objects).  This document is meant to address the issue of change surrounding device profiles in EdgeX going forward – specifically when can a device profile (or its sub-elements such as device resources) be added, modified or removed.
+
+### Summary of Device Profile Rules
+These rules will be implemented in core metadata on device profile API calls.
+
+- A device profile can be added anytime
+- Device resources or device commands can be added to a device profile anytime
+- Attributes can be added to a device profile anytime
+- A device profile can be removed or modified when the device profile is not associated to a device or provision watcher
+    - this includes modifying any field (except identifiers like names and ids)
+    - this includes changes to the array of device resources, device commands
+    - this includes changes to attributes (of device resources)
+    - even when a device profile is associated to a device or provision watcher, fields of the device profile or device resource can be modified when the field change will not affect the behavior of the system (example: the description field has no effect on the system).
```

Comment:
How does a developer know which fields change the behavior of the system and which don't?

Reply:
@tonyespy - we can list them all in the ADR (but that gets us back to the complexity again in the last document).  I would prefer we address this issue in documentation updates that we make once we've implemented.

Reply:
I've provided two additional bullets to clarify but hoping this doesn't take us back down the path of detailing out each property and when it can/cannot be changed as I fear it might.

Reply:
Thanks for adding the two additional bullets, without which and an end-user would've been left to their own devices to figure out which fields change the behavior of the system. Although I still feel like the rules we've ended up with are still a bit overly complicated, I think this is the best we're going to get, so I will go ahead and approve.